### ISSUE TITLE
Reapply Combat during Boss encounter. (BB #1163)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#1163]
This issue was migrated from bitbucket.
**Original Reporter:** Selphius
**Original Date:** 30.11.2015 14:30:44 GMT+0000
**Original Priority:** critical
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1163

<hr>

I think most people and developers know about this problem, but since it isn't reported here I will report it.

How it works:
1. The raid starts boss encounter, let's say Illidan. 
2. Player fucked up and died by Illidan's Eye Beam. 
3. He is ressurected by Combat Ress by druid. 
4. He stand up AND he ISN'T in COMBAT. It won't reapply until the player takes action to attack the boss or heal player in combat. Which makes it a Critical bug, cause ressurected player without combat can just stand back and be casting ressurection on fallen comrades, making the boss a joke not an hard encounter. It does work like that for EVERY encounter, no matter how easy or hard it is.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** desteny
**Original Date:** 30.11.2015 16:30:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1163/#comment-23688816

<hr>

there is a method something like "SetZoneInCombat()" for Cpp Scripts, this in missing in a lot of scripts.

i have implemented log ago a more generic way to set all players in a instance in combat if the targeted mob is a boss. (boss flag in DB) its quite easy.

if you (oder the team) think this is a good solution i will make a pull request (or direct commit) soon.

please give me a feedback.
thanks

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Selphius
**Original Date:** 30.11.2015 19:29:34 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1163/#comment-23693182

<hr>

Doesn't "SetZoneInCombat()" only work for "Aggro Pull"? 
I remember it worked like that on Trinity Core, that's why some people were thinking about alternatives like [Combat Pulse](http://www.wowhead.com/spell=52865/combat-pulse), but I guess it's only WoTLK Spell.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Areradon
**Original Date:** 30.11.2015 22:31:45 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1163/#comment-23697298

<hr>

As i remember SetZoneInCombat get called from Updated AI every sec. so it should fix the bug.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** desteny
**Original Date:** 01.12.2015 23:34:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1163/#comment-23729956

<hr>

Its like Arerdon says, but for me its annoing to rewrite all scripts to fix this issue.
its a repeated logic, and some kind of "code smell" for me, for this i would offer my solution if the feedback is positive, i have some time in two weeks i think i will commit it on a separated branch or on a pull request

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 26.12.2015 05:06:31 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1163/#comment-24325947

<hr>

[Core/Creatures] Add support for periodic aggro pulses.
                 This puts every player in the zone in combat and on the creature's
                 threat list.

```
             Add BossAI parent class to make use of this in _EnterCombat and
             _Reset methods.

             Fixes #1163
```

→ <<cset d2164124ecc7>>
